digitalocean/resource_digitalocean_spaces_bucket_object_test.go

```diff
@@ -0,0 +1,865 @@
+  bucket = digitalocean_spaces_bucket.object_bucket.name
+  key = "test-key"
+}
+`, testAccDigitalOceanSpacesBucketObject_TestRegion, randInt)
+}
+
+func testAccDigitalOceanSpacesBucketObjectConfigSource(randInt int, source string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "object_bucket" {
+  region = "%s"
+  name   = "tf-object-test-bucket-%d"
+}
+
+resource "digitalocean_spaces_bucket_object" "object" {
+  region       = digitalocean_spaces_bucket.object_bucket.region
+  bucket       = digitalocean_spaces_bucket.object_bucket.name
+  key          = "test-key"
+  source       = "%s"
+  content_type = "binary/octet-stream"
+}
+`, testAccDigitalOceanSpacesBucketObject_TestRegion, randInt, source)
+}
+
+func testAccDigitalOceanSpacesBucketObjectConfig_withContentCharacteristics(randInt int, source string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "object_bucket" {
+  region = "%s"
+  name   = "tf-object-test-bucket-%d"
+}
+
+resource "digitalocean_spaces_bucket_object" "object" {
+  region           = digitalocean_spaces_bucket.object_bucket.region
+  bucket           = digitalocean_spaces_bucket.object_bucket.name
+  key              = "test-key"
+  source           = "%s"
+  content_language = "en"
+  content_type     = "binary/octet-stream"
+  website_redirect = "http://google.com"
+}
+`, testAccDigitalOceanSpacesBucketObject_TestRegion, randInt, source)
+}
+
+func testAccDigitalOceanSpacesBucketObjectConfigContent(randInt int, content string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "object_bucket" {
+  region = "%s"
+  name   = "tf-object-test-bucket-%d"
+}
+
+resource "digitalocean_spaces_bucket_object" "object" {
+  region  = digitalocean_spaces_bucket.object_bucket.region
+  bucket  = digitalocean_spaces_bucket.object_bucket.name
+  key     = "test-key"
+  content = "%s"
+}
+`, testAccDigitalOceanSpacesBucketObject_TestRegion, randInt, content)
+}
+
+func testAccDigitalOceanSpacesBucketObjectConfigContentBase64(randInt int, contentBase64 string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "object_bucket" {
+  region = "%s"
+  name   = "tf-object-test-bucket-%d"
+}
+
+resource "digitalocean_spaces_bucket_object" "object" {
+  region         = digitalocean_spaces_bucket.object_bucket.region
+  bucket         = digitalocean_spaces_bucket.object_bucket.name
+  key            = "test-key"
+  content_base64 = "%s"
+}
+`, testAccDigitalOceanSpacesBucketObject_TestRegion, randInt, contentBase64)
+}
+
+func testAccDigitalOceanSpacesBucketObjectConfig_updateable(randInt int, bucketVersioning bool, source string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "object_bucket_3" {
+  region = "%s"
+  name   = "tf-object-test-bucket-%d"
+
+  versioning {
+    enabled = %t
+  }
+}
+
+resource "digitalocean_spaces_bucket_object" "object" {
+  region = digitalocean_spaces_bucket.object_bucket.region
```

Comment:
```suggestion
  region = digitalocean_spaces_bucket.object_bucket_3.region
```

Missed this one as the test is failing for other reasons.